### PR TITLE
Boilerplate Onboarding

### PR DESCRIPTION
**Pull request by dustman9000**
[OSD-5643](https://issues.redhat.com/browse/OSD-5643)
This onboards cloud-ingress-operator w/ boilerplate. 
All code changes are related to getting golangci-lint error-free.

### CODE REVIEW

**Comment by lisa** [comment]
@dustman9000 I begin my on-call shift next week, so please be advised that I will not have an excess of time to review the PR.

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go [hunk 1]
```diff
@@ -228,15 +225,19 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 					// spec that was generated based on the ApplicationIngress, and the fields checked are immutable,
 					// the actual default IngressController must be deleted
 					reqLogger.Info("Static Spec and Status do not match for default IngressController, deleting")
-					r.client.Delete(context.TODO(), ingressController)
+					if err := r.client.Delete(context.TODO(), ingressController); err != nil {
+						reqLogger.Error(err, "Error deleting IngressController")
+					}
 					return reconcile.Result{Requeue: true}, nil
```
Comment (2uasimojo) [inline pkg/controller/publishingstrategy/publishingstrategy_controller.go:231]: Do we not want to return the err here?

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go [hunk 2]
```diff
@@ -228,15 +225,19 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
...
 			} else {
 				// Since the default IngressController CR spec does not match the desired IngressController
 				// spec that was generated based on the ApplicationIngress, and the fields checked are immutable,
 				// the IngressController must be deleted
 				reqLogger.Info(fmt.Sprintf("Static Spec does not match for for IngressController %s, deleting", ingressName))
-				r.client.Delete(context.TODO(), ingressController)
+				if err := r.client.Delete(context.TODO(), ingressController); err != nil {
+					reqLogger.Error(err, "Error deleting IngressController")
+				}
 				return reconcile.Result{Requeue: true}, nil
```
Comment (2uasimojo) [inline pkg/controller/publishingstrategy/publishingstrategy_controller.go:241]: ditto

File: pkg/controller/apischeme/apischeme_controller.go [hunk 3]
```diff
@@ -303,6 +298,11 @@ func SetAPISchemeStatus(crObject *cloudingressv1alpha1.APIScheme, reason, messag
 		message,
 		utils.UpdateConditionNever)
 	crObject.Status.State = ctype
+
+	err := r.client.Status().Update(context.TODO(), crObject)
+	if err != nil {
+		log.Error(err, "Error updating cr status")
+	}
```
Comment (2uasimojo) [inline pkg/controller/apischeme/apischeme_controller.go:305]: I get that you made this logically the same as it was before, which is cool, but perhaps add a TODO to bubble this error up to the caller? Especially in the "Success" branch (L242), I would think we would want to react differently if we couldn't update the status.

File: pkg/controller/sshd/sshd_controller.go [hunk 4]
```diff
@@ -616,5 +614,8 @@ func (r *ReconcileSSHD) SetSSHDStatus(cr *cloudingressv1alpha1.SSHD, message str
 	cr.Status.State = state
 	cr.Status.Message = message
 
-	r.client.Status().Update(context.TODO(), cr)
+	err := r.client.Status().Update(context.TODO(), cr)
+	if err != nil {
+		log.Error(err, "Error updating cr status")
+	}
```
Comment (2uasimojo) [inline pkg/controller/sshd/sshd_controller.go:620]: Similar here.

File: pkg/utils/infrastructure.go [hunk 5]
```diff
@@ -55,5 +55,5 @@ func GetPlatformType(kclient client.Client) (*configv1.PlatformType, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &infra.Status.Platform, nil
+	return &infra.Status.PlatformStatus.Type, nil
```
Comment (2uasimojo) [inline pkg/utils/infrastructure.go:58]: [✓](https://github.com/openshift/api/blob/bae60f47eed768bde64c38e43c809c8275b31bb3/config/v1/types_infrastructure.go#L56)

**Review by 2uasimojo — COMMENTED** [review]
Mechanically this all looks correct to me, nice work.

Thank you for leaving TODOs in the Makefile for post-prow-config cleanup.

A couple of nonblocking comments inline. To be clear, not suggesting those changes be made here, just suggesting leaving TODOs for them.

/lgtm

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go (hunk 1 above)
Comment (dustman9000) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go:231]: Possibly, I wanted to preserve the existing behavior (ignoring an error). I can add a TODO here.

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go (hunk 2 above)
Comment (dustman9000) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go:241]: Will add a TODO.

File: pkg/controller/apischeme/apischeme_controller.go (hunk 3 above)
Comment (dustman9000) [reply pkg/controller/apischeme/apischeme_controller.go:305]: Possibly, but I wanted to preserve existing behavior. Will add a TODO.

File: pkg/controller/sshd/sshd_controller.go (hunk 4 above)
Comment (dustman9000) [reply pkg/controller/sshd/sshd_controller.go:620]: Will add a TODO.

**Comment by 2uasimojo** [comment]
Thanks, new TODOs look good. I think this is ready to go.

/lgtm

File: Makefile [hunk 6]
```diff
@@ -1,30 +1,43 @@
...
+include boilerplate/generated-includes.mk
 
-# Include shared Makefiles
-include project.mk
-include standard.mk
-include functions.mk
+SHELL := /usr/bin/env bash
 
-default: generate-syncset gobuild
+default: generate-syncset go-build
```
Comment (lisa) [inline Makefile:5]: This is overwriting the `default` target set deep in the included makefiles.

**Review by lisa — CHANGES_REQUESTED** [review]
Aside from the non-portable code within the imported boilreplate codebase (which is best addressed in that repository) this looks fine, save for the default make target in `/Makefile`.

File: Makefile (hunk 6 above)
Comment (dustman9000) [reply Makefile]: Thanks, I've now removed this.

**Comment by 2uasimojo** [comment]
/test coverage

That's odd...

**Comment by 2uasimojo** [comment]
There should be nothing weird about this codecov query, and it passed before the most recent commit, which really shouldn't have affected anything. Code-wise, this

/lgtm

...but obvs the CI failure needs :eyes:

**Comment by 2uasimojo** [comment]
coverage had a [similar glitch in c-am-o last night](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_configure-alertmanager-operator/144/pull-ci-openshift-configure-alertmanager-operator-master-coverage/1369070412670963712). It seemed to be temporary. Not sure what's going on here.

**Review by lisa — APPROVED** [review]

**Comment by lisa** [comment]
With the Makefile change done this looks okay. As before, we can address portability elsewhere, and thanks to boilerplate, this will receive those changes.

/approve
/lgtm
/pony

**Pull request merged**